citrination_client/client.py

```diff
@@ -238,6 +238,93 @@ def upload_file(self, file_path, data_set_id, root_path=None):
+            }
+        }
+        r = requests.post(url, data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
+    def matched_file_count(self, dataset_id, glob=".", is_dir=False):
+        list_result = self.list(dataset_id, glob, is_dir)
+        if isinstance(list_result["files"], list):
+            return len(list_result["files"])
+        else:
+            return list_result
+
+    def move(self, dataset_id, original_path, target_path, is_dir=False):
+        return self._move_or_copy(dataset_id, "move", original_path, target_path, is_dir)
+
+    def copy(self, dataset_id, original_path, target_path, is_dir=False):
+        return self._move_or_copy(dataset_id, "copy", original_path, target_path, is_dir)
+
+    def _move_or_copy(self, dataset_id, move_or_copy, original_path, target_path, is_dir=False):
+        """
+        Move a file within the "filesystem" of a dataset on Citrination.
+        :param dataset_id: The ID of the dataset in which the move will occur
+        :param move_or_copy: A string equal to either "move" or "copy", indicating which action should be performed
+        :param original_path: The path (to a file or to a directory) to move
+        :param target_path: The location to move the file(s)
+        :param is_dir: Boolean indicating whether or not you are moving a directory
+        """
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/' + move_or_copy
+        data = {
+            "move": {
+                "from": original_path,
+                "to": target_path,
+                "is_dir": is_dir
+            }
+        }
+        r = requests.post(url, data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
+    def get_matched_dataset_files(self, dataset_id, glob, is_dir=False, latest=True):
+        """
+        Retrieves URLs for the files matched by a glob or a path to a directory
+        in a given dataset.
+
+        :param dataset_id: The id of the dataset to retrieve files from
+        :param glob: A regex used to select one or more files in the dataset
+        :param is_dir: A flag used to indicate that the glob should be matched against the start of the paths in the dataset (simulates directory matching)
+        :param latest: A boolean flag indicating that results should be limited to reporting files from the latest dataset version
+        :return: The response object, or an error message object if the request failed
+        """
+        data = {
+            "download_request": {
+                "glob": glob,
+                "is_dir": is_dir,
+                "latest": latest
+            }
+        }
+        url = self.api_url + '/datasets/' + str(dataset_id) + '/download_files'
+        r = requests.post(url,data=json.dumps(data), headers=self.headers)
+        return json.loads(r.content.decode('utf-8'))
+
 
     def get_dataset_files(self, dataset_id, latest = False):
```

Comment:
It might help to change the name of the boolean to something like `lateset_version_only`.